components/brave_new_tab_ui/components/default/braveToday/useScrollIntoView.ts

```diff
@@ -5,26 +5,28 @@
 
 import * as React from 'react'
 
-export default function useScrollIntoView (shouldScrollIntoView: boolean) {
+export default function useScrollIntoView<T extends HTMLElement = HTMLAnchorElement> (shouldScrollIntoView: boolean) {
   // If we need to scroll the article in to view after render,
   // do so after the element has been mounted.
-  const cardRef = React.useRef<HTMLAnchorElement>(null)
+  const cardRef = React.useRef<T>(null)
   React.useEffect(() => {
     if (shouldScrollIntoView && cardRef.current) {
       cardRef.current.scrollIntoView({ block: 'center' })
+    } else if (shouldScrollIntoView) {
+      console.warn('Brave News: attempted to scroll to a card that was not rendered')
     }
-  }, []) // empty array so only runs on first render
+  }, [cardRef.current]) // only re-run if the ref element changesa
```

Comment:
mutating a ref doesn't trigger a re-render and therefore useEffect doesn't re-run. It's an unnecessary dependency. 

here's a [test](https://codesandbox.io/s/xenodochial-snowflake-hhgr6?fontsize=14&hidenavigation=1&theme=dark) that shows when useEffect actually runs and it's only right after a render.

Reply:
hmm that's a good point! But, what if ref is an html element. It changing from `null` to an element or another element would necessarily cause a render to change that element, causing the effect to then be parsed again and (I think) run because by that time hopefully ref.current has changed.

Reply:
It matters when `ref.current` is updated and that happens at a specific time for element refs. It's after the render and before useEffect. It happening after render means it's updated after the dependency array is computed, and a render isn't naturally caused after it's updated.

example cycle: 

```
initial: ___render__ __useEffect__
prop_change: ___render__ __useEffect__
```

this is where the update of ref happens.
```
initial: __ref:null__ ___render__ __ref:el__ __useEffect__
```

Reply:
hmm I think what I'm saying is that the intention behind the `ref.current` dependency is to make sure it re-runs when the element reference changes. It behaves differently from a primitive reference since I imagine an element change is always the result of a re-render.

Here's my modification of that test: https://codesandbox.io/s/q-60476155-refcurrent-in-useeffect-forked-q5gd9?file=/index.js